tuning/data/data_processors.py

```diff
@@ -398,6 +382,68 @@ def _execute_data_handlers(
               {",".join([e.name for e in DataHandlerType])}'
         )
 
+    def _concatenate_dict_datasets(
+        self, all_datasets: List[Union[DatasetDict, IterableDatasetDict]]
+    ) -> Union[DatasetDict, IterableDatasetDict]:
+        all_keys = [set(d.keys()) for d in all_datasets]
+        first_keys = all_keys[0]
+        if not all(keys == first_keys for keys in all_keys):
+            raise ValueError(
+                f"DatasetDicts do not have matching keys. Found keys: {all_keys}"
+            )
+
+        concatenated_dict = {}
+        for key in first_keys:
+            key_datasets = [d[key] for d in all_datasets]
+            maybe_align_datasets(key_datasets)
+            concatenated = datasets.concatenate_datasets(key_datasets)
+            if isinstance(concatenated, IterableDataset):
+                concatenated = resolve_iterable_dataset_features(concatenated)
+            concatenated_dict[key] = concatenated
+
+        return DatasetDict(concatenated_dict)
+
+    def _concatenate_simple_datasets(
```

Comment:
rename `_concatenate_datasets`

Reply:
`_concatenate_datasets` is already used as the name of the wrapper function

Reply:
does `_concatenate_flat_datasets` sound good?